src/h5web/dataset-visualizer/models.ts

```diff
@@ -8,5 +8,6 @@ export enum Vis {
 
 export interface DimensionMapping {
   x: number;
-  y: number;
+  y?: number;
+  slicingIndices: Record<number, number>;
 }
```

Comment:
Ok, this is where it is a bit tricky and where we may want to think really hard: the structure of the mapping object really dictates how components are created, how it should be updated and how to slice the data afterwards.

I have chosen the above structure (**which is greatly open for discussion**) where the index of the dimension that will be used for the visualisation (let's call it the data dimension) is stored as the field `x`. The other dimensions need then to be sliced and it is the job of `slicingIndices` to keep the indices used for the slicing. It is a record where the keys are all dimensions indices **except the data dimension** and the values are the indices along which the slicing should be made.

To give an example, for a 2D dataset in a `LineVis`: we can have
```
{ 
  x: 0
  slicingIndices: {1: slicingIndexForDim1}
}
```
or
```
{ 
  x: 1
  slicingIndices: {0: slicingIndexForDim0}
}
```

If we go to 3D datasets, we will have the following examples:
```
{ 
  x: 1
  slicingIndices: {0: slicingIndexForDim0, 2: slicingIndexForDim2}
}
```
or
```
{ 
  x: 2
  slicingIndices: {0: slicingIndexForDim0, 1: slicingIndexForDim1}
}
```
etc.